projects/kc63_preprocessing/reporting_period/add_reporting_period.py

```diff
@@ -0,0 +1,48 @@
+import re
+import pandas as pd
+from pathlib import Path
+from typing import Iterable
+
+YEAR_SUFFIX_RE = re.compile(r"(\d{4})$")
+
+
+def extract_year_from_filename(filename: str) -> int:
+    stem = Path(filename).stem
+    m = YEAR_SUFFIX_RE.search(stem)
+    if not m:
+        raise ValueError(f"Filename '{filename}' does not end with a 4-digit year")
+    year = int(m.group(1))
+    return year
+
+
+def build_reporting_period(year: int) -> str:
+    return f"{year-1}-{year}"
+
+
+def iter_csv_files(input_dir: Path) -> Iterable[Path]:
+    for p in sorted(input_dir.glob("*.csv")):
+        if p.is_file():
+            yield p
+
+
+def add_reporting_period_to_file(input_dir: Path, output_dir: Path) -> None:
+    year = extract_year_from_filename(input_dir.name)
+    reporting_period = build_reporting_period(year)
+    df = pd.read_csv(input_dir, low_memory=False)
+    df["reporting_period"] = reporting_period
+    df.to_csv(output_dir, index=False)
```

Comment:
Suggestion: readability; it reads like the inputs are directory paths, but I think they're file paths. How about:

```suggestion
def add_reporting_period_to_file(input_file: Path, output_file Path) -> None:
    year = extract_year_from_filename(input_file.name)
    reporting_period = build_reporting_period(year)
    df = pd.read_csv(input_file, low_memory=False)
    df["reporting_period"] = reporting_period
    df.to_csv(output_file, index=False)
```

Reply:
updated